compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeSceneMediator.uikit.kt

```diff
@@ -162,12 +199,61 @@ internal class ComposeSceneMediator(
         )
     }
 
-    private val delegate: SkikoUIViewDelegate by lazy {
-        SkikoUIViewDelegateImpl(
-            { scene },
-            interopContext,
-            densityProvider,
-        )
+    private val touchesDelegate: TouchesDelegate by lazy {
+        object : TouchesDelegate {
+            override fun pointInside(point: CValue<CGPoint>, event: UIEvent?): Boolean =
+                point.useContents {
+                    val position = this.toDpOffset().toOffset(densityProvider())
+                    !scene.hitTestInteropView(position)
+                }
+
+            override fun onTouchesEvent(view: UIView, event: UIEvent, phase: UITouchesEventPhase) {
+                scene.sendPointerEvent(
+                    eventType = phase.toPointerEventType(),
+                    pointers = event.touchesForView(view)?.map {
+                        val touch = it as UITouch
+                        val id = touch.hashCode().toLong()
+                        val position = touch.offsetInView(view, densityProvider().density)
+                        ComposeScenePointer(
+                            id = PointerId(id),
+                            position = position,
+                            pressed = touch.isPressed,
+                            type = PointerType.Touch,
+                            pressure = touch.force.toFloat(),
+                            historical = event.historicalChangesForTouch(
+                                touch,
+                                view,
+                                densityProvider().density
+                            )
+                        )
+                    } ?: emptyList(),
+                    timeMillis = (event.timestamp * 1e3).toLong(),
+                    nativeEvent = event
+                )
+            }
+        }
+    }
+
+    val renderDelegate by lazy {
+        object : RenderDelegate {
+            override fun retrieveInteropTransaction(): UIKitInteropTransaction =
+                interopContext.retrieve()
+
+            override fun render(canvas: Canvas, targetTimestamp: NSTimeInterval) {
+                // The calculation is split in two instead of
+                // `(targetTimestamp * 1e9).toLong()`
+                // to avoid losing precision for fractional part
+                val integral = floor(targetTimestamp)
+                val fractional = targetTimestamp - integral
+                val secondsToNanos = 1_000_000_000L
+                val nanos = integral.roundToLong() * secondsToNanos + (fractional * 1e9).roundToLong()
```

Comment:
I asked in the prev PR to extract such calculation from the render function, it seems that it was missed.

Reply:
I don't know where to extract it. I can't find a good place.

Reply:
You can merge the proposed change from [this branch](https://github.com/JetBrains/compose-multiplatform-core/tree/es/time-interval-to-nano-extraction)

Reply:
Done